src/main/java/uk/gov/hmcts/reform/notificationservice/service/NotificationMessageParser.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.reform.notificationservice.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.servicebus.MessageBody;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import uk.gov.hmcts.reform.notificationservice.exception.InvalidMessageException;
+import uk.gov.hmcts.reform.notificationservice.model.request.incomming.NotificationMsg;
+
+import java.io.IOException;
+import java.util.List;
+
+@Service
+public class NotificationMessageParser {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NotificationMessageParser.class);
+    private static final String ERROR_CAUSE = "Message Binary data is null";
+
+    private final ObjectMapper objectMapper;
+
+    public NotificationMessageParser(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    public NotificationMsg parse(MessageBody messageBody) {
+        try {
+            NotificationMsg notificationMsg =
+                objectMapper.readValue(getBinaryData(messageBody), NotificationMsg.class);
+            LOGGER.info(
+                "Parsed notification message, Zip File Name: {}, Error Code: {}, Error Description: {} "
+                    + "Jurisdiction: {}, PO Box: {}, Service: {}, Document Control Number: {}",
+                notificationMsg.zipFileName,
+                notificationMsg.errorCode,
+                notificationMsg.errorDescription,
+                notificationMsg.jurisdiction,
+                notificationMsg.poBox,
+                notificationMsg.service,
+                notificationMsg.documentControlNumber
+            );
+
+            return notificationMsg;
+        } catch (IOException exc) {
+            LOGGER.error("Notification queue message parse error", exc);
+            throw new InvalidMessageException(exc);
```

Comment:
When we log an error and then throw exception, we end up having two error log entries, both representing the same event. It would be best to not log anything at this point, but make sure the exception that we throw will be logged instead (plus include enough information in the exception message).

Reply:
I think we should log sth as this is a runtime exception. we do not need to log exceptions itself.?

Reply:
I think it's safe to assume that all exceptions that represent an error situation will be logged. It's up to us, anyway.

Reply:
removed.